### DISCUSSION TITLE
model mocking mongo

### DISCUSSION THREAD

**Discussion by navyaharish** [jmcdo29/testing-nestjs#1389] [General]
How do i  use two models here :

provide: getModelToken('Cat')

**Comment by jmcdo29** [comment]
I'm not sure I follow. Do you need to mock two different `@InjectModel()`s?

**Reply by navyaharish** [reply]
Yes

**Reply by jmcdo29** [reply]
You'd need to set up two custom providers. Say you have `@InjectModel('Cat')` and `@InejctModel('Dog')`, then you'd need to do something like
```ts
providers:  [
	{
		provide: getModelToken('Cat'),
		useValue: catModelMock,
	},
	{
		provide: getModelToken('Dog'),
		useValue: dogModelMock,
	}
]
```

**Reply by navyaharish** [reply]
That worked, thanks! :)
